controllers/configurationpolicy_controller.go

```diff
@@ -3279,29 +3286,130 @@ func (r *ConfigurationPolicyReconciler) sendComplianceEvent(instance *policyv1.C
 	return r.Create(context.TODO(), event)
 }
 
-// convertPolicyStatusToString to be able to pass the status as event
-func convertPolicyStatusToString(plc *policyv1.ConfigurationPolicy) string {
+// defaultComplianceMessage looks through the policy's Compliance and CompliancyDetails and formats
+// a message that can be used for compliance events recognized by the framework.
+func defaultComplianceMessage(plc *policyv1.ConfigurationPolicy) string {
 	if plc.Status.ComplianceState == "" || plc.Status.ComplianceState == policyv1.UnknownCompliancy {
 		return "ComplianceState is still unknown"
 	}
 
-	result := string(plc.Status.ComplianceState)
+	defaultTemplate := `
+	{{- range .Status.CompliancyDetails -}}
+	  ; {{ if (index .Conditions 0) -}}
+	    {{- (index .Conditions 0).Type }} - {{ (index .Conditions 0).Message -}}
+	  {{- end -}}
+	{{- end }}`
+
+	// `Must` is ok here because an invalid template would be caught by tests
+	t := template.Must(template.New("default-msg").Parse(defaultTemplate))
+
+	var result strings.Builder
+
+	result.WriteString(string(plc.Status.ComplianceState))
+
+	if err := t.Execute(&result, plc); err != nil {
+		log.Error(err, "failed to execute default template", "PolicyName", plc.Name)
+
+		// Fallback to just returning the compliance state - this will be recognized by the framework,
+		// but will be missing any details.
+		return string(plc.Status.ComplianceState)
+	}
+
+	return result.String()
+}
+
+// customComplianceMessage uses the custom template in the policy (if provided by the user) to
+// format a compliance message that can be used by the framework. If an error occurs with the
+// template, the default message will be used, appended with details for the error. If no custom
+// template was specified for the current compliance, then the default message is used.
+func (r *ConfigurationPolicyReconciler) customComplianceMessage(plc *policyv1.ConfigurationPolicy) string {
+	customTemplate := plc.Spec.CustomMessage.Compliant
+
+	if plc.Status.ComplianceState != policyv1.Compliant {
+		customTemplate = plc.Spec.CustomMessage.NonCompliant
+	}
+
+	defaultMessage := defaultComplianceMessage(plc)
+
+	// No custom template was provided for the current situation
+	if customTemplate == "" {
+		return defaultMessage
+	}
+
+	customMessage, err := r.doCustomMessage(plc, customTemplate, defaultMessage)
+	if err != nil {
+		return fmt.Sprintf("%v (custom message template failure: %v)", defaultMessage, err.Error())
```

Comment:
(Optional) To me, swapping the wording seems slightly clearer.
```suggestion
		return fmt.Sprintf("%v (failure processing the custom message: %v)", defaultMessage, err.Error())
```